jobs/periodic/sync-to-webflow/sync-courses.ts

```diff
@@ -0,0 +1,172 @@
+        const startDate = moment(lecture.start);
+        if (earliestDate == null || startDate.isBefore(earliestDate)) {
+            earliestDate = moment(lecture.start);
+        }
+    }
+    return earliestDate;
+}
+
+function getCouseDuration(subcourse: Subcourse): number {
+    let duration = 0;
+    for (const lecture of subcourse.lecture) {
+        duration += lecture.duration;
+    }
+    return duration;
+}
+
+function listAppointments(subcourse: Subcourse): string {
+    let appointments = [];
+    for (const lecture of subcourse.lecture) {
+        const startDate = moment(lecture.start);
+        appointments.push(startDate.format('dddd, DD. MMMM YYYY, HH:mm [Uhr]'));
+    }
+    return appointments.join('\n');
+}
+
+function courseToDTO(subcourse: Subcourse): CourseDTO {
+    const startDate: Moment = getStartDate(subcourse) || moment();
+    const courseDTO: CourseDTO = {
+        ...emptyMetadata,
+
+        name: subcourse.course.name,
+        // TODO add databaseid field ot cms
+        databaseid: `${subcourse.id}`, // We are using a string to be safe for any case.
+
+        description: subcourse.course.description,
+        instructor: generateHost(subcourse.subcourse_instructors_student),
+        aboutinstructor: '', // TODO
+        subheading: subcourse.course.outline,
+
+        startingdate: startDate.toISOString(),
+        weekday: startDate.format('dddd'),
+        courseduration: `${getCouseDuration(subcourse)} min`, // TODO: maybe this can be done in a nicer way. Maybe with the help of moment?
+        lessoncount: subcourse.lecture.length,
+        time: startDate.format('HH:mm'),
+        appointments: listAppointments(subcourse),
+
+        category: subcourse.course.category, // TODO reference
+        link: '', // TODO Wird noch gebraucht?
+        maxparticipants: subcourse.maxParticipants,
+        actualparticipants: subcourse.subcourse_participants_pupil.length,
+        subject: subcourse.course.subject, // TODO reference
+
+        mingrade: subcourse.minGrade,
+        maxgrade: subcourse.maxGrade,
+        priorknowledge: false, // TODO Ich kann die Daten nicht finden
+        agegroup: '', // TODO Wir haben die Daten nicht
+        level: '', // TODO
+
+        imagecopyrightinfo: subcourse.course.imageKey,
+        thumbnail: {
+            fileId: 'testId', // reference
+            url: subcourse.course.imageKey,
+            alt: '',
+        },
+        image: {
+            fileId: 'testId', // reference
+            url: subcourse.course.imageKey,
+            alt: '',
+        },
+    };
+    courseDTO.slug = hash(courseDTO);
+    return courseDTO;
+}
+
+export default async function syncCourses(logger: Logger): Promise<void> {
+    logger.info('Start course sync');
+    const webflowCourses = await getCollectionItems<WebflowMetadata>(collectionId, courseDTOFactory);
+    const subCourses = await prisma.subcourse.findMany({
+        where: { course: { courseState: courseState.allowed } },
```

Comment:
Use IS_PUBLIC_SUBCOURSE() as filter, this is not sufficient (it will also show cancelled, not yet published and past subcourses)